src/mutation_data.cpp

```diff
@@ -641,7 +641,21 @@ void dream::load( const JsonObject &jsobj )
 
 bool trait_display_sort( const trait_id &a, const trait_id &b ) noexcept
 {
-    return a->name() < b->name();
+    if( a->get_display_color() > b->get_display_color() ) {
+        return true;
+    }
+    if( a->get_display_color() < b->get_display_color() ) {
+        return false;
+    }
+
+    if( a->name() < b->name() ) {
+        return true;
+    }
+    if( a->name() > b->name() ) {
+        return false;
+    }
+
+    return false;
```

Comment:
You could skip this condition and `return false` directly (or replace it with `else` and remove the last `return`)

Reply:
I could also just remove that name check altogether and just do:
return a->name() < b->name();

I wrote it that way to keep the code style in the method consistent, I'm not worried about the additional cost of 2 clock cycles.

Reply:
@CodeBandit I was more thinking about readability than optimization in that case.
But I think @jbytheway's suggestion is better anyway.

Reply:
@enaantd I 100% agree it's a far elegant line of code. Unfortunately, it didn't match the same sort order criteria when I built and unit tested